### PR TITLE
[core] clean up core package & update public docs with typescript syntax 

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/core/ts/MobileCore.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ const RCTAEPCore: IMobileCore = NativeModules.AEPCore;
 const MobileCore: IMobileCore = {
   /**
    * Returns the version of the AEPCore extension
-   * @param  {string} Promise a promise that resolves with the extension version
+   * @return  {string} Promise a promise that resolves with the extension version
```
Comment (cacheung) [inline packages/core/ts/MobileCore.ts:46]: * @return {string} Promise that resolves with the extension version

File: packages/core/ts/MobileCore.ts [hunk 2]
```diff
@@ -182,7 +182,7 @@ const MobileCore: IMobileCore = {
    *  button or a certain article was viewed.
    *
    *  note: when using the Adobe Analytics extension, calling this API will increment page views
-   * @param  {String?} state containing the name of the state to track
+   * @param  {String?} action containing the name of the action to track
    * @param  {{ string: string }?} contextData containing context data to attach on this hit
```
Comment (cacheung) [inline packages/core/ts/MobileCore.ts:186]: * @param {Record<string, string>} contextData containing context data to attach on this hit

File: packages/core/ts/Signal.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const RCTAEPSignal: ISignal = NativeModules.AEPSignal;
 const Signal: ISignal = {
   /**
    * Returns the version of the AEPSignal extension
-   * @param  {string} Promise a promise that resolves with the extension version
+   * @return  {string} Promise a promise that resolves with the extension version
```
Comment (cacheung) [inline packages/core/ts/Signal.ts:24]: @return {string} Promise that resolves with the extension version

File: packages/core/ts/MobileCore.ts [hunk 4]
```diff
@@ -115,7 +115,7 @@ const MobileCore: IMobileCore = {
    * more verbose than the current {@link LogLevel} set from {@link #setLogLevel(LogLevel)}
    * then the message is not printed.
    *
-   * @param mode the {@link LogLevel} used to print the message
+   * @param logLevel the {@link LogLevel} used to print the message
    * @param tag used to identify the source of the log message
    * @param message the message to log
    */
```
Comment (cacheung) [inline packages/core/ts/MobileCore.ts:121]: In Set the Adobe Mobile Privacy status below
* @param {PrivacyStatus} PrivacyStatus to be set to the SDK

File: packages/core/ts/MobileCore.ts (hunk 1 above)
Comment (yangyansong-adbe) [reply packages/core/ts/MobileCore.ts:46]: `{string} Promise`  is the return type, and the following `a promise ...` is the description. I think we'd better keep the current description.

File: packages/core/ts/Signal.ts (hunk 3 above)
Comment (yangyansong-adbe) [reply packages/core/ts/Signal.ts:24]: We don't need to update it, the reason is the same as ->  https://github.com/adobe/aepsdk-react-native/pull/157#discussion_r837675134

**Review by cacheung — APPROVED** [review]

**Pull request merged**